pkg/dsl/task/task.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 CUE Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package task provides a registry for tasks to be used by commands.
+package task
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/errors"
+)
+
+// Context provides context for running a task.
+type Context struct {
```

Comment:
Maybe consider a more descriptive name and not conflict with the go built-in "context"

Reply:
done